.github/workflows/ci.yml

```diff
@@ -0,0 +1,92 @@
+name: Kubernetes-sigs/krew CI
+on: 
+  push:
+    branches: 
+      - '*'
+    tags: 
+      - 'v*.*.*'
+  pull_request:
+    branches: 
+      - '*'
+  release:
+    types: [published, created, edited]
+
+jobs:
+
+  build:
+    name: Build and Tags
+    runs-on: ubuntu-latest
+    steps:
+
+    - name: Set up Go 1.13
+      uses: actions/setup-go@v1
+      with:
+        go-version: 1.13
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+    
+    - name: Set GOPATH
+      # temporary fix
+      # see https://github.com/actions/setup-go/issues/14
+      run: |
+        echo "##[set-env name=GOPATH;]$(dirname $GITHUB_WORKSPACE)"
+        echo "##[add-path]$(dirname $GITHUB_WORKSPACE)/bin"
+      shell: bash
+    
+    - name: Install gox
+      run: |
+        go mod download
+        hack/install-gox.sh\
+
+    - name: Install dependencies
+      run: go mod tidy && git diff --no-patch --exit-code
+      
+    - name: Verify code quality
+      run: |
```

Comment:
If one of the scripts in this multi-line run statement fails, the following statements will be executed regardless.

In this case, this is actually fine (and even desired), so no need for action ;)

Reply:
well actually... I this _is_ a problem, because the success status of the whole block will be the return value of the last line. So an error in one of the previous lines will not propagate and halt the pipeline. That's not good :/

Reply:
Yeah I'm almost never cool with that.

Similarly I can guess if you make a typo in a $variable_name, it won't error out. This is [bash strict mode](http://redsymbol.net/articles/unofficial-bash-strict-mode/) all over again.

For these situations, we should prefer creating hack/*.sh scripts that adhere to the same set of practices.

Reply:
Yeah, putting everything in bash scripts sounds like a manageable solution.

Reply:
Well, I read some solutions, run those files in differents steps, run it with a bash statement like this https://github.com/kubernetes-sigs/krew-index/pull/437#pullrequestreview-344230430, or finally create a new file *.sh and run all those steps. I would like to know your opinion. Then I could develop it :)

Reply:
Yeah we can split all scripts into different `steps[]`. The fact that you have a && in the name suggests that they are different steps.

Bash carries out exit code of "last" command (in this case it would be the "only" command if we do 1 command per step).

But when it doesn't make sense to split multiple commands into two different steps, we should not shy away from using `set -ex`.

Reply:
`set -u` seems also advisable (see discussion in krew-index)

Reply:
> or finally create a new file *.sh and run all those steps

I personally think that we have too many bash scripts already. So if we can avoid adding another one that is only needed for the CI, I would prefer that.

Reply:
+1 separating each script into a step (just like they were in .travis.yml) is much better than a single opaque script.

Reply:
I've separated some steps, and others used `set -euo pipefail` as `krew-index`.